cmake/CompilerOddities.cmake

```diff
@@ -60,11 +62,11 @@ macro(Set_Compiler_Specific_Flags)
 
     message(FATAL_ERROR "Cannot determine noexcept fladg on std::ios_base::failure.  See log.")
   elseif(COMPILER_HAS_NOEXCEPT_WHAT_ON_EXCEPTIONS_V1) 
-    add_definitions(-DCOMPILER_MODIFIER_ON_EXCEPTION_WHAT=noexcept)
+    add_compiler_source_define(COMPILER_MODIFIER_ON_EXCEPTION_WHAT=noexcept)
   elseif(COMPILER_HAS_NOEXCEPT_WHAT_ON_EXCEPTIONS_V2)
-    add_definitions(-DCOMPILER_MODIFIER_ON_EXCEPTION_WHAT=_NOEXCEPT)
+    add_compiler_source_define(DCOMPILER_MODIFIER_ON_EXCEPTION_WHAT=_NOEXCEPT)
```

Comment:
should remove `D`, will modify this in my branch.

Reply:
Yes, that makes sense.